apps/docs/src/content/docs/guide/oauth.md

```diff
@@ -0,0 +1,105 @@
+
+After you've registered your application, it is important that you take note of the client secret somewhere safe. If the client secret is to ever leak, it is important that you regenerate it to ensure the security of your authorized users. If your client secret or access tokens are found exposed in the wild, your application may be disabled without prior notice.
+
+## Getting authorization
+
+Once the user is ready to authorize your application, you need to construct a URL to redirect them to. The authorization URL for Modrinth is `https://api.modrinth.com/_internal/oauth/token`. Supply the following query parameters:
+
+| Query parameter | Description                                                                               |
+|-----------------|-------------------------------------------------------------------------------------------|
+| `response_type` | In Modrinth this always needs to be `code`, since only code grants are supported          |
+| `client_id`     | The application identifier found in the settings                                          |
+| `scope`         | The permissions you need access to                                                        |
+| `state`         | A mechanism to prevent certain attacks. Explained further below. Recommended but optional |
+| `redirect_uri`  | The URI the user is redirect to after finishing authorization                             |
+
+The scope identifiers are currently best found in the backend source code located at [`apps/labrinth/src/models/v3/pats.rs`]. The scope parameter is an array of scope identifiers, seperated by a plus sign (`+`).
+
+The redirect URI is the endpoint on your server that will receive the code which can eventually be used to act on the user's behalf. For security reasons the redirect URI used has to be allowlisted in your application settings. The redirect will contain the following query parameters:
+
+| Query parameter | Description                                        |
+|-----------------|----------------------------------------------------|
+| `code`          | The code that can be exchanged for an access token |
+| `client_id`     | Your client id                                     |
+| `redirect_uri`  | The redirect URI which was used                    |
+| `grant_type`    | Always `authorization_code` in Modrinth            |
+
+In the next chapter we will exchange the authorization code for an access token, but before that we have to discuss this `state` parameter. [CSRF] (Cross-site request forgery), and [clickjacking] are security vulnerabilities that you're recommended to protect against. In OAuth2 this is usually done with the `state` parameter. When the user initiates a request to start authorization, you include a `state` which is unique to this request. This can, for example, be saved in localStorge or a cookie. When the redirect URI is called, you verify that the `state` is the same. Using `state` is optional, but recommended.
```

Comment:
Do we implement this? It's not in the previous docs at all.

If yes, it should be moved up to be after the table that actually includes the state variable information, not tacked on as an afterthought at the end of the section.